src/port/port.tcl

```diff
@@ -3927,6 +3927,128 @@ proc action_echo { action portlist opts } {
+permissions for group and others:
+"
+    if {[llength $paths] > 0} {
+        set allow [$macports::ui_options(questions_yesno) $msg2 "allow" $paths {y} 0 "Do you want to allow this?"]
+        if {$allow == 1} {
+            return 0
+        }
+    }
+
+    # CREATE GIT SOURCE BY CLONING MACPORTS-PORTS
+    if {[catch {system "${macports::findBinary git} clone --depth 100 $url $path"} result]} {
+        ui_debug $::errorInfo
+        return 1
+    }
+
+    # TODO: CHANGE PERMISSIONS OF $PATH TO ROOT
+
+    # COPY INDEXFILE FROM OLD SOURCE TO GIT SOURCE
+    set default_source_url [lindex $sources_default 0]
+    set indexfile [macports::getindex $default_source_url]
+    if {[catch {file copy -force $indexfile $path} error]} {
+        ui_debug $::errorInfo
+        ui_error "file copy: $error"
+        return -code error "Copying Rsync IndexFile to new git source failed"
+    }
+
+    # EDIT SOURCES.CONF TO USE GIT SOURCE
+    # Construct sed command
+    set cmdline $portutil::autoconf::sed_command -E "s/(^rsync)/#\1/g"
+    # Create temporary sources.conf.bump.XXXXXX
+    if {[catch {set tmpfile [mkstemp "/tmp/sources.conf.sed.XXXXXX"]} error]} {
+        ui_debug $::errorInfo
+        ui_error "mkstemp: $error"
+        return -code error "mkstemp failed"
+    }
+    # Extract the Tcl Channel number
+    set tmpfd [lindex $tmpfile 0]
+    # Set tmpfile to only the file name
+    set tmpfile [join [lrange $tmpfile 1 end]]
+    # Get sources.conf attributes
+    set attributes [file attributes $indexfile]
+
+    # Direct sed command output to tempfile
+    lappend cmdline "<${default_source_url}" ">@$tmpfd"
+
+    # Run sed command and write to sources.conf.bump
+    if {[catch {exec -ignorestderr -- {*}$cmdline} error]} {
+        ui_debug $::errorInfo
+        ui_error "sed: $error"
+        file delete "$tmpfile"
+        close $tmpfd
+        return -code error "sed sed(1) failed"
+    }
+
+    # Replace sources.conf with sources.conf.bump
+    if {[catch {move -force $tmpfile $indexfile} error]} {
+        ui_debug $::errorInfo
+        ui_error "bump: $error"
+        file delete "$tmpfile"
+        return -code error "bump overwrite failed"
+    }
+
+    # Restore sources.conf attributes
+    file attributes $portfile {*}$attributes
+
+    # syncing ports tree.
+    if {![info exists options(ports_selfupdate_no-sync)]} {
+        if {$comp > 0} {
+            # updated portfiles potentially need new base to parse - tell sync to try to
+            # use prefabricated PortIndex files and signal if it couldn't
+            lappend optionslist no_reindex 1 needed_portindex_var needed_portindex
+        }
+        try {
+            mportsync $optionslist
+        }  catch {{*} eCode eMessage} {
+            return -code error "Couldn't sync the ports tree: $eMessage"
+        }
+    }
+
+    ui_msg "Your /opt/local/etc/macports/sources.conf has been updated to use this repository."
+}
```

Comment:
You may want a `return yes/1/true/truthyvalue` here.

You should probably also take a second look at your return values in this function.